### PR TITLE
Origins - Mynt FE

### PR DESCRIPTION
**Pull request by charles-dr**

### CODE REVIEW

**Review by soulBit — CHANGES_REQUESTED** [review]
Looks good other than the few small requested changes

File: src/app/hooks/useDollarValue.ts [hunk 1]
```diff
@@ -26,3 +26,26 @@ export function useDollarValue(asset: Asset, weiAmount: string) {
...
+  const dollars = useCachedAssetPrice(asset, Asset.USDT);
+
+  const value = useMemo(() => {
+    const { decimals } = AssetsDictionary.get(asset);
+    if ([Asset.USDT, Asset.DOC, Asset.RDOC].includes(asset)) {
+      return weiAmount;
+    } else {
+      return bignumber(weiAmount)
+        .mul(dollars.value)
+        .div(10 ** (decimals + 2))
```
[the remark this answers is not in the document]
Comment (soulBit) [reply src/app/hooks/useDollarValue.ts]: Agree with this, please implement as tiltom suggested

File: src/app/pages/OriginsLaunchpad/index.tsx [hunk 2]
```diff
@@ -4,15 +4,13 @@ import { useTranslation } from 'react-i18next';
 import { translations } from 'locales/i18n';
 import { Header } from 'app/components/Header';
 import { Footer } from 'app/components/Footer';
-import { Dashboard } from './pages/Dashboard';
+// import { Dashboard } from './pages/Dashboard';
 import { SalesDay } from './pages/SalesDay';
 /* undo once Fish contract has active sale tier reset to 0 */
```
Comment (tiltom) [inline src/app/pages/OriginsLaunchpad/index.tsx:9]: This comment is not true anymore and you don't need the import below it as well.

File: src/utils/dateHelpers.ts [hunk 3]
```diff
@@ -2,3 +2,12 @@ import dayjs from 'dayjs';
 
 export const getNextMonday = () =>
   dayjs().utc().startOf('week').add(1, 'week').day(1).format('MMMM D');
+
+export const timestampToDateTimeString = (timestamp: number) =>
```
Comment (tiltom) [inline src/utils/dateHelpers.ts:6]: You still use another method (`timestampToString`) in `useGetSaleInformation` for this. Use this method instead.

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/app/pages/OriginsLaunchpad/pages/SalesDay/index.tsx [hunk 4]
```diff
@@ -30,12 +30,14 @@ export const SalesDay: React.FC<ISalesDayProps> = ({ tierId, saleName }) => {
             tierId={tierId}
```
[the remark this answers is not in the document]
Comment (soulBit) [reply src/app/pages/OriginsLaunchpad/pages/SalesDay/index.tsx]: Images will be replaced once provided by Tzu

File: src/utils/dateHelpers.ts (hunk 3 above)
Comment (business-helper) [reply src/utils/dateHelpers.ts:6]: **timestampToString** and **timestampToDateTimeString**  return the different result.
**timestampToString** - returns month and date only
**timestampToDateTimeString** - returns monty, date, hours and minutes.
I moved **timestampToString** to the **dateHelpers** and imported it in **useGetSaleInformation**.

**Review by soulBit — COMMENTED** [review]
Changes look good, just waiting on the Mynt images from Tzu - once those are provided and updated in this PR I will approve merging

**Comment by soulBit** [comment]
@business-helper a few points raised by QA during review:

1. The "Token Sale End Time" should include ordinal suffix on the date (e.g. "4th Nov"). I also suggest changing this to UTC locale (screenshot below shows "3 Nov" as the end date, whereas on my machine it shows "4 Nov").
2. We need to update "Buy MYNT" button opacity when disabled (reduce to 0.5 opacity).
3. We need to update "Buy MYNT" button validation to handle this case: when user selects an asset in their wallet where they have 0 available balance, and then enters a number in the amount input manually, they are able to click the buy button and the transaction will fail since the funds are not available in the wallet. See screenshot for an example

image

File: src/utils/dateHelpers.ts [hunk 5]
```diff
@@ -11,3 +11,9 @@ export const timestampToDateTimeString = (timestamp: number) =>
     hour12: false,
     minute: 'numeric',
   });
+
+export const timestampToString = (timestamp: number) =>
```
Comment (tiltom) [inline src/utils/dateHelpers.ts:15]: I would rename it to `timestampToDateString` as you already have `timestampToDatetimeString` in the file and `timestampToString` is too ambigous.

File: src/app/components/UserAssets/index.tsx [hunk 6]
```diff
@@ -236,7 +236,7 @@ function AssetRow({
   const dollarValueMynt = useDollarValueMynt(tokens);
   const _dollarValue = useMemo(
```
Comment (tiltom) [inline src/app/components/UserAssets/index.tsx:237]: Rename it to something else, `_dollarValue` is really lazy as you have a variable named `dollarValue` two lines above it. I would say it could be `assetDollarValue`, what do you think @soulBit ?

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/app/components/UserAssets/index.tsx (hunk 6 above)
Comment (soulBit) [reply src/app/components/UserAssets/index.tsx:237]: tiltom's suggestion sounds good here 👍🏻

File: src/app/components/PromotionModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+const shouldModalBeVisible = () => !local.getItem(SESSION_KEY);
+
+export const PromotionModal: React.FC = () => {
+  const { t } = useTranslation();
+  const [show, setShow] = useState(shouldModalBeVisible());
+  const history = useHistory();
+  useEffect(() => setShow(shouldModalBeVisible()), []);
+
+  const handleClose = () => {
```
Comment (tiltom) [inline src/app/components/PromotionModal/index.tsx:24]: Use `useCallback`.

File: src/app/components/PromotionModal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    local.setItem(SESSION_KEY, '1');
+    setShow(false);
+  };
+
+  const handleSaleOpen = () => {
```
Comment (tiltom) [inline src/app/components/PromotionModal/index.tsx:29]: Use `useCallback`.

File: src/app/components/PromotionModal/index.tsx [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+    history.push('/origins');
+  };
+
+  return (
+    <Dialog
+      isOpen={show}
+      onClose={handleClose}
+      isCloseButtonShown={true}
+      canEscapeKeyClose={true}
+      className={cn('tw-w-full tw-max-w-6xl tw-p-6', styles.dialog)}
```
Comment (tiltom) [inline src/app/components/PromotionModal/index.tsx:40]: Use `classNames` instead of `cn`.

File: src/app/components/PromotionModal/index.tsx [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+    >
+      <div className="tw-font-light tw-text-center tw-w-full tw-mx-auto">
+        <div
+          className="tw-font-bold tw-text-center tw-mb-6"
+          style={{ fontSize: '25px' }}
```
Comment (tiltom) [inline src/app/components/PromotionModal/index.tsx:45]: Why do you have it as inline style? I would create a new selector in `index.module.scss` and you can use it then in `className` (also, add the existing classnames to the new selector by `@apply`).

File: src/app/components/PromotionModal/index.tsx [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+              src={imgLargeNFT}
+              alt="MYNT"
+              className="tw-border-8 tw-border-gray-9 tw-rounded-3xl"
+            />
+          </div>
+          <div className="tw-w-6/12 tw-mr-32">
+            <DialogWrapper className="tw-justify-evenly tw-h-full">
+              <div
+                className="tw-font-bold tw-text-center tw-mb-6"
+                style={{ fontSize: '25px' }}
```
Comment (tiltom) [inline src/app/components/PromotionModal/index.tsx:61]: Why do you have it as inline style? I would create a new selector in index.module.scss and you can use it then in className (also, add the existing classnames to the new selector by `@apply`).

File: src/app/components/PromotionModal/styled.ts [hunk 12]
```diff
@@ -0,0 +1,62 @@
+import styled from 'styled-components';
```
Comment (tiltom) [inline src/app/components/PromotionModal/styled.ts:1]: We don't use `styled components` anymore. Rewrite it to `CSS modules` (see some `index.module.scss` for inspiration).

File: src/app/pages/OriginsLaunchpad/pages/SalesDay/pages/BuyStep/components/BuyDialog/components/BuySection/styled.ts [hunk 13]
```diff
@@ -26,13 +26,14 @@ export const BuyButton = styled.button<IBuyButtonProps>`
   text-transform: uppercase;
   padding: initial;
 
-  &:hover {
+  &:not([disabled]):hover {
```
Comment (tiltom) [inline src/app/pages/OriginsLaunchpad/pages/SalesDay/pages/BuyStep/components/BuyDialog/components/BuySection/styled.ts:29]: I probably didn't notice it before but we don't use `styled components` in our code anymore, we use `CSS modules` instead. Rewrite this file to `index.module.scss` instead and you can take a look at some of ours `index.module.scss` for inspiration.

Also, rewrite all of the `styled.ts` files used in this PR to `CSS modules`.

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/app/pages/OriginsLaunchpad/pages/SalesDay/index.module.scss [hunk 14]
```diff
@@ -1,15 +1,13 @@
...
-export const TitleImage = styled.img`
+.titleImage {
   width: 55px;
   height: 54px;
-`;
+}
 
-export const TitleContent = styled.div`
+.titleContent {
   font-size: 2.5rem;
```
Comment (tiltom) [inline src/app/pages/OriginsLaunchpad/pages/SalesDay/index.module.scss:7]: We have a rule that we try to use `@apply` to as many properties as possible and we write new ones only if they don't have the value we need in Tailwind.

So in this case it would like this: 
`@apply tw-font-semibold tw-leading-none tw-tracking-normal tw-ml-8 tw-uppercase;
font-size: 2.5rem;`

As you can see, only `font-size` has a custom value (and sometimes you find out there is no custom value in your styles and you can delete a selector). Please, go through all of your `index.module.scss` in this PR and adjust them.

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/app/pages/OriginsLaunchpad/pages/SalesDay/index.module.scss (hunk 14 above)
Comment (business-helper) [reply src/app/pages/OriginsLaunchpad/pages/SalesDay/index.module.scss]: I updated the styling as you specified.

File: src/locales/en/translation.json [hunk 15]
```diff
@@ -301,76 +301,86 @@
       }
     },
     "saleDay": {
-      "title": "{{token}} Origins Pre-Sale",
+      "title": "{{token}}-Token Bootstrap Event on Origins ",
```
Comment (john-light) [inline src/locales/en/translation.json:304]:
```suggestion
      "title": "{{token}} Bootstrap Event on Origins ",
```

File: src/locales/en/translation.json [hunk 16]
```diff
@@ -301,76 +301,86 @@
...
       "engageWalletScreen": {
-        "dialogTitle": "Welcome to the {{token}} Origins Pre-Sale",
+        "dialogTitle": "Welcome to the {{token}} Bootstrap Event on Origins",
         "buttonText": "Connect Wallet",
         "dialogFooter1": "Engage a browser, mobile or hardware wallet connected to the <0>RSK Mainnet</0>",
         "dialogFooter2": "Don’t have a wallet? Sovryn recommends <0>Liquality wallet</0>"
       },
       "accessCodeVerificationStep": {
-        "dialogTitle": "Welcome to the {{token}} Origins Pre-Sale",
+        "dialogTitle": "Welcome to the {{token}}-Token Bootstrap Event on Origins ",
```
Comment (john-light) [inline src/locales/en/translation.json:312]:
```suggestion
        "dialogTitle": "Welcome to the {{token}} Bootstrap Event on Origins ",
```

File: src/locales/en/translation.json [hunk 17]
```diff
@@ -301,76 +301,86 @@
...
+        "alert": {
+          "closed": "{{token}}-Token bootstrap event already finished.",
```
Comment (john-light) [inline src/locales/en/translation.json:314]:
```suggestion
          "closed": "{{token}} bootstrap event already finished.",
```

File: src/locales/en/translation.json [hunk 18]
```diff
@@ -301,76 +301,86 @@
...
+          "active": "You can now participate in the {{token}}-Token bootstrap event.",
```
Comment (john-light) [inline src/locales/en/translation.json:315]:
```suggestion
          "active": "You can now participate in the {{token}} bootstrap event.",
```

File: src/locales/en/translation.json [hunk 19]
```diff
@@ -301,76 +301,86 @@
...
+          "notYetOpen": "The {{token}}-Token public bootstrap event will open at {{time}}, get your {{depositToken}} ready!"
```
Comment (john-light) [inline src/locales/en/translation.json:316]:
```suggestion
          "notYetOpen": "The {{token}} public bootstrap event will open at {{time}}, get your {{depositToken}} ready!"
```

File: src/locales/en/translation.json [hunk 20]
```diff
@@ -301,76 +301,86 @@
...
-            "5": "FISH tokens purchased as part of this private pre-sale, will vest linearly for approximately 10 months from the time they are claimed.",
-            "6": "IMPORTANT - You will need to actively claim your FISH tokens on the Origins interface, from August 28 at 14:00 UTC. Your vesting period will begin only once you have claimed your tokens.",
-            "7": "A FISH AMM pool will be available on Sovryn. FISH can be used to vote in BabelFish governance as soon as it begins to vest and after BabelFish governance has been launched.",
-            "8": "For any questions, please ask the community in Discord or Telegram channels."
+            "1": "You have been whitelisted for the Blockchain Whispers' Private {{token}} pre-sale! Welcome and thanks for your support!",
+            "2": "It is highly recommended to send funds from an address directly owned by you (EOA).",
+            "3": "For any questions, please ask the community in Discord or Telegram channels",
+            "4": "The price of 100 {{token}} Tokens is 1 SOV",
```
Comment (john-light) [inline src/locales/en/translation.json:333]:
```suggestion
            "4": "The price of 100 {{token}} tokens is 1 SOV",
```

File: src/locales/en/translation.json [hunk 21]
```diff
@@ -301,76 +301,86 @@
...
+            "6": "The sale ends in 7 Days.",
+            "7": "Once you complete the purchase of the {{token}} Token it will be visible on the Portfolio Section.",
+            "8": "There will be no vesting schedule for the purchased Token. So you can transfer Tokens once the sale ends."
           },
           "checkboxText": "I have read and understand that I am responsible for my own Sovrynity",
           "submitButtonText": "I Understand"
         },
         "publicSale": {
-          "title": "Important Information about the the FISH Public Pre-Sale Process and Terms",
+          "title": "Important Information about the {{token}}-Token Bootstrap Event",
```
Comment (john-light) [inline src/locales/en/translation.json:343]:
```suggestion
          "title": "Important Information about the {{token}} Bootstrap Event",
```

File: src/locales/en/translation.json [hunk 22]
```diff
@@ -301,76 +301,86 @@
...
-            "5": "Half the FISH purchased will vest (be liquid) upon completion of the pre-sale, the other half will vest linearly for approximately 10 consecutive months from the time they are claimed.",
-            "6": "IMPORTANT - You will need to actively claim your FISH tokens on the Origins interface, from August 28 at 14:00 UTC. Your vesting period will begin only once you have claimed your tokens.",
-            "7": "The FISH tokens that are immediately vested will be available for withdrawal from August 28, at 14:00 UTC.",
-            "8": "A FISH AMM pool will be available on Sovryn. FISH can be used to vote in BabelFish governance as soon as it begins to vest and after BabelFish governance has been launched.",
-            "9": "For any questions, please ask the community in Discord or Telegram channels."
+            "1": "Welcome to the {{token}}-Token Bootstrap Event. Thank you for your participation!",
```
Comment (john-light) [inline src/locales/en/translation.json:345]:
```suggestion
            "1": "Welcome to the {{token}} Bootstrap Event. Thank you for your participation!",
```

File: src/locales/en/translation.json [hunk 23]
```diff
@@ -301,76 +301,86 @@
...
+            "2": "We advise you to participate in this sale with a wallet solely and directly owned by you (EOA). ",
+            "3": "For any questions, please ask the community in our <0>Discord</0> or <1>Telegram</1> channels.",
+            "4": "The price of 100 {{token}} is 1 SOV. Other assets will be converted to SOV at the current exchange rate prior to the purchase.",
+            "5": "There will be no vesting schedule for the purchased Token. You can transfer purchased {{token}}-tokens once the sale is concluded.",
```
Comment (john-light) [inline src/locales/en/translation.json:349]:
```suggestion
            "5": "There will be no vesting schedule for the purchased token. You can transfer purchased {{token}} once the sale is concluded.",
```

File: src/locales/en/translation.json [hunk 24]
```diff
@@ -301,76 +301,86 @@
...
+            "6": "The sale ends on 11th November 2021.",
```
Comment (john-light) [inline src/locales/en/translation.json:350]:
```suggestion
            "6": "The sale ends on 15th November 2021.",
```

File: src/locales/en/translation.json [hunk 25]
```diff
@@ -301,76 +301,86 @@
...
+            "7": "Once you receive the purchased {{token}}-Token it will be visible in your <0>Portfolio</0> section.",
```
Comment (john-light) [inline src/locales/en/translation.json:351]:
```suggestion
            "7": "Once you receive the purchased {{token}} it will be visible in your <0>Portfolio</0> section.",
```

File: src/locales/en/translation.json [hunk 26]
```diff
@@ -301,76 +301,86 @@
...
+            "8": "There will be no vesting schedule for the purchased Token. So you can transfer Tokens once the sale ends."
```
Comment (john-light) [inline src/locales/en/translation.json:352]:
```suggestion
```

Comment (john-light) [reply src/locales/en/translation.json:352]: This is the same as point (5) above

File: src/locales/en/translation.json [hunk 27]
```diff
@@ -301,76 +301,86 @@
...
         "buyDialog": {
           "enterAmount": "Enter Amount",
           "tokenReceived": "{{token}} Received",
           "transactionFee": "Transaction Fee",
           "buyButton": "Buy {{token}}",
-          "isOverMaxLimit": "You cannot deposit more than max limit"
+          "isOverMaxLimit": "You cannot deposit more than max limit",
+          "yourTotalDeposit": "Your Total Deposit",
+          "balance": "Balance",
+          "equivalentTokenAmount": "Equivalent {{token}} Token : {{amount}}"
```
Comment (john-light) [inline src/locales/en/translation.json:379]:
```suggestion
          "equivalentTokenAmount": "Estimated {{token}} to receive : {{amount}}"
```

File: src/locales/en/translation.json [hunk 28]
```diff
@@ -1320,6 +1330,14 @@
       "paragraph2": "Until then, why not try the desktop version?"
     }
   },
+  "promotionsDialog": {
+    "title": "New Event on the Origins Launchpad",
+    "promotion1_title": "MYNT-Token",
```
Comment (john-light) [inline src/locales/en/translation.json:1335]:
```suggestion
    "promotion1_title": "MYNT",
```

File: src/locales/en/translation.json [hunk 29]
```diff
@@ -1320,6 +1330,14 @@
...
+    "promotion1_1": "The MYNT-Token Bootstrap Event is now live!",
```
Comment (john-light) [inline src/locales/en/translation.json:1336]:
```suggestion
    "promotion1_1": "The MYNT Bootstrap Event is now live!",
```

**Comment by tiltom** [comment]
Title in the banner should have some `margin-top`.

image

File: src/app/pages/OriginsLaunchpad/pages/SalesDay/index.module.scss [hunk 30]
```diff
@@ -4,10 +4,5 @@
...
 .titleContent {
-  font-size: 2.5rem;
-  font-weight: 600;
-  line-height: 1;
-  letter-spacing: 0;
-  margin-left: 2rem;
-  text-transform: uppercase;
-}
\ No newline at end of file
+  @apply tw-text-4xl tw-font-semibold tw-leading-none tw-tracking-normal tw-ml-8 tw-uppercase;
```
Comment (tiltom) [inline src/app/pages/OriginsLaunchpad/pages/SalesDay/index.module.scss:7]: In cases like this, you should delete the selector and use the classes directly in the component because you don't add any custom value here.

File: src/app/pages/OriginsLaunchpad/pages/SalesDay/pages/AccessCodeVerificationStep/index.module.scss [hunk 31]
```diff
@@ -1,16 +1,11 @@
...
-  display: flex;
-  flex-direction: column;
-  justify-content: space-between;
 }
 
 .dialogTitle {
-  font-size: 1.75rem;
-  letter-spacing: 0;
-  margin-bottom: 5rem;
+  @apply tw-text-3xl tw-tracking-normal tw-mb-20;
```
Comment (tiltom) [inline src/app/pages/OriginsLaunchpad/pages/SalesDay/pages/AccessCodeVerificationStep/index.module.scss:10]: Delete the selector and use classes directly in the component.

File: src/app/pages/OriginsLaunchpad/pages/SalesDay/pages/BuyStep/components/BuyDialog/components/TxDialog/dialog.module.scss [hunk 32]
```diff
@@ -33,19 +32,15 @@
...
+    tw-w-40 tw-h-40
+    tw-border-solid tw-border-sov-white tw-overflow-hidden;
   border-radius: 1.25rem;
-  border: 5px solid #e8e8e8;
+  border-width: 5px;
   margin: 0 auto 4.375rem;
   padding: 1.3rem 2.1rem 1rem 2.1rem;
 }
 
 .WLImage {
```
Comment (tiltom) [inline src/app/pages/OriginsLaunchpad/pages/SalesDay/pages/BuyStep/components/BuyDialog/components/TxDialog/dialog.module.scss:44]: Delete the selector and use classes directly in the component.

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/app/pages/OriginsLaunchpad/pages/SalesDay/index.module.scss (hunk 14 above)
Comment (tiltom) [reply src/app/pages/OriginsLaunchpad/pages/SalesDay/index.module.scss]: It looks good now, I just found 3 selectors that can be deleted (see my latest comments).

File: src/app/pages/OriginsLaunchpad/pages/SalesDay/index.module.scss (hunk 30 above)
Comment (business-helper) [reply src/app/pages/OriginsLaunchpad/pages/SalesDay/index.module.scss:7]: Okay, I understand.

File: src/app/components/PromotionModal/index.tsx [hunk 33]
```diff
@@ -14,7 +14,7 @@ import promotionModalStyles from './index.module.scss';
 
 const SESSION_KEY = 'sovryn-promo-dialog-1'; //increment this whenever new content is added, to ensure users see it even if they viewed one previously
 
-const shouldModalBeVisible = () => !local.getItem(SESSION_KEY);
+const shouldModalBeVisible = () => !!local.getItem(SESSION_KEY);
```
Comment (tiltom) [inline src/app/components/PromotionModal/index.tsx:17]: Why did you change this? Now you basically say that the modal should be visible only if you have already seen it so it means you won't ever see it.

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by tiltom — DISMISSED** [review]

**Review by soulBit — APPROVED** [review]

**Comment by soulBit** [comment]
@bitcoinuser you can add translations for this now

**Review by soulBit — APPROVED** [review]
